src/benchmarks/micro/libraries/System.Numerics.Vectors/Perf_VectorConvert.cs

```diff
@@ -0,0 +1,232 @@
+
+        [Benchmark]
+        public Vector<long> Widen_int() => Widen<int, long>(s_valuesInt32);
+
+        [Benchmark]
+        public Vector<ulong> Widen_uint() => Widen<uint, ulong>(s_valuesUInt32);
+
+
+        // These arrays are used for the Narrow benchmarks, so they need 2 vectors per iteration
+        private static readonly double[] s_valuesDouble = ValuesGenerator.Array<double>(Vector<double>.Count * 2 * Iterations);
+        private static readonly short[] s_valuesInt16 = ValuesGenerator.Array<short>(Vector<short>.Count * 2 * Iterations);
+        private static readonly ushort[] s_valuesUInt16 = ValuesGenerator.Array<ushort>(Vector<ushort>.Count * 2 * Iterations);
+        private static readonly int[] s_valuesInt32 = ValuesGenerator.Array<int>(Vector<int>.Count * 2 * Iterations);
+        private static readonly uint[] s_valuesUInt32 = ValuesGenerator.Array<uint>(Vector<uint>.Count * 2 * Iterations);
+        private static readonly long[] s_valuesInt64 = ValuesGenerator.Array<long>(Vector<long>.Count * 2 * Iterations);
+        private static readonly ulong[] s_valuesUInt64 = ValuesGenerator.Array<ulong>(Vector<ulong>.Count * 2 * Iterations);
+
+        // The remainder are used only for Widen and same-size Convert benchmarks
+        private static readonly float[] s_valuesFloat = ValuesGenerator.Array<float>(Vector<float>.Count * Iterations);
+        private static readonly byte[] s_valuesByte = ValuesGenerator.Array<byte>(Vector<byte>.Count * Iterations);
+        private static readonly sbyte[] s_valuesSByte = ValuesGenerator.Array<sbyte>(Vector<sbyte>.Count * Iterations);
+
+        private static Vector<TTo> Convert<TFrom, TTo>(TFrom[] values) where TFrom : struct where TTo : struct
+        {
+            var input = Unsafe.As<Vector<TFrom>[]>(values);
+            var accum = Vector<TTo>.Zero;
+
+            ref Vector<TFrom> ptr = ref input[0];
+            for (int i = Iterations; i >= 0; --i)
+            {
+                accum ^= ConvertVector<TFrom, TTo>(ptr);
+                ptr = ref Unsafe.Add(ref ptr, 1);
+            }
+
+            return accum;
+        }
+
+        private static Vector<TTo> Narrow<TFrom, TTo>(TFrom[] values) where TFrom : struct where TTo : struct
+        {
+            var input = Unsafe.As<Vector<TFrom>[]>(values);
+            var accum = Vector<TTo>.Zero;
+
+            ref Vector<TFrom> ptr = ref input[0];
+            for (int i = Iterations; i >= 0; --i)
+            {
+                accum ^= NarrowVector<TFrom, TTo>(ptr, Unsafe.Add(ref ptr, 1));
+                ptr = ref Unsafe.Add(ref ptr, 2);
+            }
+
+            return accum;
+        }
+
+        private static Vector<TTo> Widen<TFrom, TTo>(TFrom[] values) where TFrom : struct where TTo : struct
+        {
+            var input = Unsafe.As<Vector<TFrom>[]>(values);
+            var accum = Vector<TTo>.Zero;
+
+            ref Vector<TFrom> ptr = ref input[0];
+            for (int i = Iterations; i > 0; --i)
```

Comment:
why two of the loops are limited by `i >= 0` while this one by `i > 0` (missing equal operator?) ?

Reply:
Just an oversight.  Thanks for catching!